mlir/test/Conversion/MemRefToLLVM/memref-to-llvm.mlir

```diff
@@ -289,6 +289,26 @@ func.func @address_space(%arg0 : memref<32xf32, affine_map<(d0) -> (d0)>, 7>) {
 
 // -----
 
+// ALL-LABEL: func @llvm_address_space_cast
+//  ALL-SAME:   (%[[ARG:.*]]: memref<f32, #llvm.address_space<3>>)
+func.func @llvm_address_space_cast(%arg0 : memref<f32, #llvm.address_space<3>>) -> memref<f32, 3 : i32> {
```

Comment:
I'm a bit confused by `#llvm.address_space`. Never seen it before. Why would we have `memref<..., #llvm.address_space<3>>` instead of `memref<..., 3>`?